branding_company/model/branding_company.py

```diff
@@ -30,6 +29,29 @@ class BrandingCompany(models.Model):
         supporting models and reports.
         """
 
+    def get_partner_branding_company(self, partner_id):
+        """Returns recordset with branding company for partner or False."""
+        if not partner_id:
+            return False
+        partner_model = self.env['res.partner']
+        partner_obj = partner_model.browse(partner_id)
+        return partner_obj and partner_obj.branding_company_id or False
```

Comment:
thanks, my example is totally wrong. So my proposal should have been `self.env['res.partner'].browse([])`, which is equivalent to your `self.env['res.partner']`